src/com/owncloud/android/operations/UpdateShareViaLinkOperation.java

```diff
@@ -56,6 +57,7 @@ public UpdateShareViaLinkOperation(long shareId) {
         mPassword = null;
         mExpirationDateInMillis = 0;
         mPublicUpload = null;
+        mPermissions = OCShare.DEFAULT_PERMISSION;
```

Comment:
There's no way to keep the current permission right now. If we don't send the permission we want in each update, it will be set to READ, have a look at https://github.com/owncloud/core/blob/7b7dfbcaf7360c88125d6ce5365146623d6c7195/apps/files_sharing/lib/API/Share20OCS.php#L370